### PR TITLE
feat: top reader badges

### PR DESCRIPTION
**Pull request by omBratteng**
## Changes

AS-510

### Preview domain
https://as-510.preview.app.daily.dev

### Badge endpoint preview
https://as-510.preview.app.daily.dev/badges/e81c38aa-a8d6-4cf4-93e5-db24ec5daf30

### Storybook preview
https://storybook-git-as-510-dailydotdev.vercel.app/?path=/story/components-topreaderbadge--default

### Profile preview
https://as-510.preview.app.daily.dev/ombratteng

### Comments preview
https://as-510.preview.app.daily.dev/posts/what-do-we-think-of-this-one--isxfepzre

### CODE REVIEW

File: packages/shared/src/components/profile/UserShortInfo.tsx
Comment (rebelchris) [inline packages/shared/src/components/profile/UserShortInfo.tsx:121]: We could have probably combined the Seperator inside the TopReaderIn, but it's fine as is

**Review by rebelchris — APPROVED** [review]
Went over it, and no particular issues on my side :) 
Let's wait for one more approval and showcase to product first.

File: packages/shared/src/components/profile/UserShortInfo.tsx
Comment (omBratteng) [reply packages/shared/src/components/profile/UserShortInfo.tsx:121]: Thought about it too, but considered not too at it might be confusing (DX wise) why the other elements has a separator in front, and not this one (even if it’s inside it)

File: packages/shared/src/components/comments/CommentContainer.tsx
Comment (ilasw) [inline packages/shared/src/components/comments/CommentContainer.tsx:157]: We should check how this behave with accessibility, now aria-label state "November 1996", but the actual content is more meaningful!

image

File: packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx
Comment (ilasw) [inline packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx:32]: Maybe we can make it easier?

```ts
const topReader = topReaders?.[0];
```

File: packages/shared/src/components/post/SquadPostAuthor.tsx
Comment (ilasw) [inline packages/shared/src/components/post/SquadPostAuthor.tsx:114]: Same as before

File: packages/shared/src/lib/image.ts
Comment (ilasw) [inline packages/shared/src/lib/image.ts:260]: Image is ~800x800 but badge is way more smaller. 
Do you think we could use a smaller one with repeat? 👀

**Review by ilasw — COMMENTED** [review]
Nice work! 
Looks good to me, just pointed some minor accessibility/performance issues

File: packages/shared/src/components/badges/TopReaderBadge.tsx
Comment (omBratteng) [reply packages/shared/src/components/badges/TopReaderBadge.tsx]: Yeah, changed to use `time`

File: packages/shared/src/components/comments/CommentContainer.tsx
Comment (omBratteng) [reply packages/shared/src/components/comments/CommentContainer.tsx:157]: It hardcodes the `aria-label` to be the content
```ts
tooltipProps['aria-label'] = content
```

File: packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx
Comment (omBratteng) [reply packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx]: Done

File: packages/shared/src/components/post/SquadPostAuthor.tsx
Comment (omBratteng) [reply packages/shared/src/components/post/SquadPostAuthor.tsx:114]: Which before?

Comment (ilasw) [reply packages/shared/src/components/post/SquadPostAuthor.tsx:114]: `aria-label` one 👀

File: packages/shared/src/components/comments/CommentContainer.tsx
Comment (ilasw) [reply packages/shared/src/components/comments/CommentContainer.tsx:157]: Yes, maybe we can add `aria-label` as`TopReaderIn` prop and (later even in a new PR) set that if `aria-label` is already set we don't want to override it?

Comment (omBratteng) [reply packages/shared/src/components/comments/CommentContainer.tsx:157]: It's from the tooltip

File: packages/shared/src/components/post/SquadPostAuthor.tsx
Comment (ilasw) [reply packages/shared/src/components/post/SquadPostAuthor.tsx:114]: Oh mb

File: packages/shared/src/components/comments/CommentContainer.tsx
Comment (omBratteng) [reply packages/shared/src/components/comments/CommentContainer.tsx:157]: But I am extending the tooltip component to take aria-label prop to use instead.

Comment (ilasw) [reply packages/shared/src/components/comments/CommentContainer.tsx:157]: Nice, thanks!

File: packages/shared/src/lib/image.ts
Comment (omBratteng) [reply packages/shared/src/lib/image.ts]: I replaced it with SVG version, so ~50% reduction in size.

Comment (ilasw) [reply packages/shared/src/lib/image.ts]: Nice!

**Review by ilasw — APPROVED** [review]
Nice!

**Pull request merged**
